README.md
# mdsanima-rt-go

![GitHub issues](https://img.shields.io/github/issues-raw/mdsanima-dev/mdsanima-rt-go?style=flat)
![GitHub closed issues](https://img.shields.io/github/issues-closed/mdsanima-dev/mdsanima-rt-go?style=flat)
![GitHub](https://img.shields.io/github/license/mdsanima-dev/mdsanima-rt-go?style=flat)
![Lines of code](https://img.shields.io/tokei/lines/github/mdsanima-dev/mdsanima-rt-go?style=flat)
![GitHub top language](https://img.shields.io/github/languages/top/mdsanima-dev/mdsanima-rt-go)

Cross Platform Application for Calculating Render Time.

## Testing

This is a base application build on `Windows` `Android` and `Linux`.

All build is a debug mode.
This is a not production build only for testing.

`Linux Ubuntu 20.04`

<img width="960" alt="mdsanima rt go-0 3 0-linux64-debug" src="https://user-images.githubusercontent.com/3817871/130751221-8093fb70-de62-4392-83a9-8cb5e5e42edd.png">

`Microsoft Windows 10`

<img width="960" alt="mdsanima rt go-0 3 0-windows64-debug" src="https://user-images.githubusercontent.com/3817871/130751295-f150a805-cea8-45f7-8cb5-9d532a81f259.png">

`Samsung S7 Edge Android armeabi-v7a`

| ![mdsanimartgo-030-armeabi-v7a-debug_samsung_s7edge_splash][1] | ![mdsanimartgo-030-armeabi-v7a-debug_samsung_s7edge_main][2] | ![omdsanimartgo-030-armeabi-v7a-debug_samsung_s7edge_notification][3] |
|---|---|---|

[1]: https://user-images.githubusercontent.com/3817871/130751388-167051bd-88f0-4743-a088-42dc4f1ac64c.jpg
[2]: https://user-images.githubusercontent.com/3817871/130751452-478777d8-8852-4fdd-acae-e7bfc2fe5b89.jpg
[3]: https://user-images.githubusercontent.com/3817871/130751489-7007822f-ae2b-4448-9960-5fcb14cbe537.jpg

## Requirement

Run this command in **wsl-1** and **PowerShell** on the `Windows 10`:

```shell
python3.8 -m pip install -r requirements.txt
```

If kivy does not detect opengl add this code to `setting.py`:

```python
import os
os.environ['KIVY_GL_BACKEND'] = 'angle_sdl2'
```

## Run `standard-version`

First run **standard-version** on **wsl** to generate `CHANGELOG.md`
and bumping verion in `__init__.py` and `spec_windows_version.rc` files:

```shell
standard-version --skip.commit --skip.tag
```

Only bump release as minor and check:

```shell
standard-version --skip.changelog --skip.commit --skip.tag --release-as minor --dry-run
```

Add bumping files and commiting:

```shell
git add . && git commit -m "chore(release): 0.1.0"
```

Create tag:

```shell
standard-version --skip.bump --skip.changelog --commit-all --sign
```

## Build Windows `.exe`

Run on `PowerShell` in Windows:

```PowerShell
cd J:\github-mdsanima-dev\mdsanima-rt-go
.\spec_windows_build.cmd
```

## Build Linux

Run on `shell` in `ubuntu_20_04` Virtual Box:

```shell
sudo su
cd /media/sf_mdsanima-rt-go
sudo python3.8 -m PyInstaller \
    --distpath /media/sf_mdsanima-rt-go/dist/linux \
    --workpath /media/sf_mdsanima-rt-go/.build \
    --onefile /media/sf_mdsanima-rt-go/spec_linux_onefile.spec
```

## Build Android `armeabi-v7a` `arm64-v8a` `x86` `x86_64`

Connect phone to pc and run on `PowerShell` this command:

```PowerShell
cd C:\adb
.\adb.exe devices
```

If List of devices attached is empty run thic command:

```PowerShell
.\adb.exe kill-server
.\adb.exe devices
```

Now shuld show devices

Run this on `wsl` to copy all files in to `home` directory:

```shell
cd /home/mdsanima/mdsanima-rt-go/
rm -r * && cp -r /mnt/j/github-mdsanima-dev/mdsanima-rt-go/* .
```

Before building if requirements on buildozer.spec updated:

```shell
sudo buildozer android clean
```

Build and deploy:

```shell
sudo buildozer --profile armeabi-v7a -v android debug deploy run logcat
sudo buildozer --profile arm64-v8a -v android debug
sudo buildozer --profile x86 -v android debug
sudo buildozer --profile x86_64 -v android debug
```

Adb install:

```shell
adb install -r /home/mdsanima/mdsanima-rt-go/dist/android/mdsanima.rt.go-0.3.0-armeabi-v7a-debug.apk
```

Copy `.apk` to `dist` folder:

```shell
cp dist/android/mdsanima.rt.go-0.1.0-armeabi-v7a-debug.apk \
    /mnt/j/github-mdsanima-dev/mdsanima-rt-go/dist/android/
```

## Connect With Me

Hi there, I'm Marcin Różewski aka [MDSANIMA](https://mdsanima.com).
These are my social media, check it out please. Thanks.

![GitHub followers](https://img.shields.io/github/followers/mdsanima?style=social)
![Twitter Follow](https://img.shields.io/twitter/follow/toudajew?style=flat-square)
![Twitter Follow](https://img.shields.io/twitter/follow/str9led?style=flat-square)
![Twitter Follow](https://img.shields.io/twitter/follow/mdsanima?style=flat-square)
![YouTube Channel Subscribers](https://img.shields.io/youtube/channel/subscribers/UCB5na2BRwrnwx00LCspbG5Q?style=social)
![YouTube Channel Views](https://img.shields.io/youtube/channel/views/UCB5na2BRwrnwx00LCspbG5Q?style=social)

## License

Cross Platform App `MDSANIMA RT GO` is released under the terms of
[Apache-2.0 License](https://github.com/mdsanima-dev/mdsanima-rt-go/blob/master/LICENSE)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.
See [standard-version](https://github.com/conventional-changelog/standard-version)
and `README.md` file for commit guidelines. Also you should check [conventional-changelog](https://github.com/mdsanima/conventional-changelog) for more detail.

Docomentation for **mdsanima-rt-go** Aplication is available
at [docs mdsanima-dev](https://mdsanima-dev.github.io/mdsanima-rt-go/) be sure to check it.

## [0.3.0](https://github.com/mdsanima-dev/mdsanima-rt-go/compare/v0.2.0...v0.3.0)

*Date:* *2021-08-25*

### Features

* add toolbar with dark color closes [#29](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/29) ([22b9929](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/22b9929c27fd7548ac228501f26660fc2a08ab43))
* animation time code and frame rate labels ([aa00d9f](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/aa00d9f7193b2f66ec65abb38befde5baa2895a5))
* app info button and set transition direction `up` closes [#31](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/31) ([92f2e3f](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/92f2e3f9c0e9a306f57ac9afc165d9b89c850ac3))
* ASCII art images in `info app` screen closes [#46](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/46) ([abfb3b5](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/abfb3b55cff98cc46b14d56f5e37b96aca9d366b))
* back button to `main` screen closes [#42](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/42) ([39e9974](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/39e9974e2e62be87e91e8d422087409189c4ace2))
* blur bacground image on second `main` screen closes [#30](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/30) ([f295c5c](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/f295c5c89d15de4e2c45b7104d5b9fa788ffa82f))
* blured background on `info app` screen closes [#41](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/41) ([c1e03a9](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/c1e03a942a5975ff99593b1701f8a7612b9755e2))
* convert frames to timecode format ([5d11f68](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/5d11f681cf811383d2883a2c54e969367d2a2d14))
* date time start and complete progress red when no rt value ([01b45cb](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/01b45cb76911527ce26eb5bf95bacfe3097805b3))
* implement sliders values show on label ([be9e0f7](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/be9e0f75f86c0901fa88fc0a73eae94173863e1c))
* label renderi time result closes [#34](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/34) ([777a004](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/777a004dfb2e4fb265ba7f1f001aa3d77aef2712))
* render time total calculation and human friendly format ([b905e16](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/b905e161b67328afc4e3ede02fd7bb0bc78a0a1c))
* **devs:** function convert frames to time code ([2558a38](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/2558a3846c970410deccbcc6b3587dd1d273480b))
* human friendly value text render time one frame ([b1a0009](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/b1a000906afdd193cff11aa9baabd9934abbef1f))
* info version on the bottom `info app` screen closes [#40](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/40) ([9c3f8c5](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/9c3f8c5b1ab33fa1e5012947ed58ef6ebfabf7aa))
* label start and finish render time closes [#35](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/35) and [#36](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/36) ([eaf391f](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/eaf391f64a90a3f340207e95809ca0e0740ea665))
* progress bar loading animation on `welcome` screen closes [#27](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/27) ([8621e13](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/8621e13920a72bfe815e9b1e932a7cacb1328175))
* scroll view in `info app` screen closes [#43](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/43) ([a27f1bc](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/a27f1bc5ad2a450c31cb89ae0092e5f1ad4b0cde))
* sliders for render time one frame closes [#33](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/33) ([8cebb2a](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/8cebb2a8d600083d142efe464b6658d14e8ecd18))
* text fields how many frames closes [#32](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/32) ([ad92cf4](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/ad92cf4b9c6f9f43534efa63a73bbb877db40f75)), closes [#29](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/29)

### Bug Fixes

* **android:** sliders align ([5ead2a2](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/5ead2a2911e4ff9e8edc7a1a6d5e9cf536c39ae2))
* **android:** top navigation ([76f6794](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/76f67949b7d003004efc1775fab4def32f4c6616))
* app info button on the middle of screen ([6b5dfc1](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/6b5dfc1d4afd4518948f7ce14294518c9a55d848))
* background folder name ([ef4f669](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/ef4f669750c702cb799bc35511f8e2956e42c5d8))
* brightness and contrast on the bg welcome screen ([9808a78](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/9808a78af8ea872ce0bc7116797cfbe2932a39f8))
* icons on first screen and `app info` closes [#53](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/53) ([e7aa073](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/e7aa073f956f941eca85bc6b0944300f454a7c9f))
* image launch for android drop shadow white ([0054bec](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/0054bec4105cc16f16de331cad8fb1aadcac3cf1))
* **build-system:** update images and icons closes [#38](https://github.com/mdsanima-dev/mdsanima-rt-go/issues/38) ([fc67664](https://github.com/mdsanima-dev/mdsanima-rt-go/commit/fc676644cb27cf3c8f7ff28add773d48ad321f14))

src/libs/screen/info.py
"""
Calculation module. Here is a main screen render time calculation.
"""


from kivy.uix.image import Image
from kivy.uix.screenmanager import Screen
from kivy.uix.scrollview import ScrollView
from kivymd.uix.boxlayout import MDBoxLayout
from kivymd.uix.floatlayout import MDFloatLayout
from kivymd.uix.label import MDLabel
from kivymd.uix.list import MDList

from __init__ import __version__, resource_path
from config.image import get_images
from libs.screen.layout import MDSRTGO_layout


class MDSRTGO_scr_3(Screen):
    """
    Info app screen cover information of the `MDSANIMA RT GO` application.

    :param Screen: screen class kivy
    :type Screen: class
    """
    def __init__(self, **kwargs):
        super(MDSRTGO_scr_3, self).__init__(**kwargs)
        # assigning function and class to variable
        layout_mds = MDSRTGO_layout()
        layout = MDFloatLayout()
        img = get_images()

        # assigning class to variable
        img_logo_mdsanima = Image(
            source=resource_path(img[5]),
            size=('211sp','211sp'), pos=self.pos,
            size_hint_y=None,
            allow_stretch=True,
        )

        img_ascii_mdsanima = Image(
            source=resource_path(img[4]),
            size=('422sp','422sp'), pos=self.pos,
            size_hint_y=None,
            allow_stretch=True,
        )

        # draw info version background top name
        layout_mds.img_background(resource_path(img[2]))
        layout_mds.lbl_top('arrow-left', self.screen_switch, 'APP INFORMATION')
        layout_mds.lbl_info_version()
        self.add_widget(layout_mds)

        # draw all widget
        self.add_widget(layout)

        # assigning class to variable
        layout_box = MDBoxLayout(
            orientation='vertical',
            padding=[0,'60sp',0,'35sp']
        )
        scroll_view = ScrollView()
        self.list_view = MDList()

        # add widget layout
        layout_box.add_widget(scroll_view)
        scroll_view.add_widget(self.list_view)

        # assigning function to variable brake labels
        break_info = self.lbl_info_break

        # add widget layout
        self.lbl_info_top('MDSANIMA RT GO APP')
        self.lbl_info_bottom('Created by Marcin Różewski')
        break_info()
        self.lbl_info_top('GIVE ME A FEEDBACK')
        self.lbl_info_bottom('admin@app.mdsanima.com')
        break_info()
        self.lbl_info_top('DEVELOPMENT')
        self.lbl_info_bottom('We developing these application for you!')
        self.lbl_info_bottom('1.5K lines of code it written by MDSANIMA')
        break_info()
        self.list_view.add_widget(img_logo_mdsanima)
        break_info()
        self.lbl_info_top('APP INFORMATION')
        self.lbl_info_app('app name', 'MDSANIMA RT GO')
        self.lbl_info_app('license', 'Apache - 2.0')
        self.lbl_info_app('source code on', 'GitHub')
        self.lbl_info_app('release version', 'v' + __version__)
        self.lbl_info_app('first release on', 'August 19, 2021')
        self.lbl_info_app('language', 'Python v3.8.6 64-bit')
        self.lbl_info_app('framework', 'Kivy v2.0.0')
        self.lbl_info_app(' ', 'KivyMD v0.104.2')
        self.lbl_info_app('cloud provider', 'Google Cloud Platform', 'Hint')
        self.lbl_info_app(' ', 'Firebase Realtime Database', 'Hint')
        self.lbl_info_app(' ', 'Firebase Storage', 'Hint')
        self.lbl_info_app('distribution', 'Cross Platform')
        self.lbl_info_app(' ', 'Microsoft Windows 10')
        self.lbl_info_app(' ', 'Linux Ubuntu 20.04')
        self.lbl_info_app(' ', 'Android API 21 - 29')
        self.lbl_info_app(' ', 'macOSX - Coming Soon', 'Hint')
        self.lbl_info_app(' ', 'iOS - Coming Soon', 'Hint')
        break_info()

        # initial variable info text
        info_vfx_studio = (
            "Let me introduce myself. I'm in charge of head of VFX and Game "
            + "Development for MDSANIMA. We've been working on VFX and Game "
            + "Development for the past 10 years now."
            + "\n\n"
            + "We love Houdini Sidefx so much. We create visual effect like "
            + "smoke, water, cloth simulation, procedural 3D modeling and "
            + "much more we do with these software. If you want to, you can "
            + "make digital asset for Unreal Engine 4 tweak parameter of your "
            + "asset directly in game engine. In this way, you can make "
            + "perfect game with only this two excellent software."
            + "\n\n"
            + "Of course we love Blender too. In these software, we make "
            + "3D models, sculpt and render. Also we have experienced on "
            + "Substance Painter UDIM workflow, Zbrush, Coat 3D, Nuke, Black "
            + "Magic Davinci Resolve, Black Magic Fusion and many many more "
            + "great software to do making games, cinematic, commercial, as "
            + "well as app development."
            + "\n\n"
            + "We recently discovered a great software to do rendering and "
            + "layout 3d stuff called Clarisse iFX. This software can handle "
            + "off the limit the biggest 3D scenes ever, without any display "
            + "problems. Handle over 10 trylion polygon and no more lagging "
            + "on 3D viewport, no more LOD - level of detail."
        )

        info_app_devs = (
            "Our studio also experienced in App Development. We're "
            + "programming in Python, VEX, C#, Java, PHP, HTML and SQL. "
            + "Development Framework Xamarin, Kivy, KivyMD, PySide2, PyQt5 "
            + "and more. Our favorite programing software is Visual Studio "
            + "Code work with GIT."
            + "\n\n"
            + "We'r expanded our experience in Cloud Computing like Linux "
            + "Ubuntu Spot Instance in AWS, S3, Microsoft Azure, Oracle, "
            + "Google Cloud Platform with Realtime Database in Firebase."
            + "\n\n"
            + "We really like command line rendering, so much fun!"
        )

        # add widget layout
        self.lbl_info_top('MDSANIMA VFX Studio')
        self.lbl_info_mdsanima(info_vfx_studio, 'justify', '520sp')
        break_info()
        self.lbl_info_top('APP DEVS')
        self.lbl_info_mdsanima(info_app_devs, 'justify', '255sp')
        break_info()
        self.lbl_info_top('OUR CLIENTS')
        self.lbl_info_bottom('admin@app.mdsanima.com')
        break_info()
        self.list_view.add_widget(img_ascii_mdsanima)

        # draw all widget
        self.add_widget(layout_box)

    def screen_switch(self):
        self.manager.current = 'scr_2'
        self.manager.transition.direction = 'right'

    def lbl_info_top(self, text_info: str):
        info_top = MDLabel(
            text=text_info,
            halign='center',
            height='15sp',
            #font_style='H6',
            size_hint_y=None,
            theme_text_color='Custom',
            text_color=[.2510,.5529,.9765,1]
        )
        self.list_view.add_widget(info_top)

    def lbl_info_bottom(self, text_info: str, height: str='30sp'):
        info_bottom = MDLabel(
            text=text_info,
            halign='center',
            height=height,
            size_hint_y=None,
            theme_text_color='Secondary'
        )
        self.list_view.add_widget(info_bottom)

    def lbl_info_break(self):
        info_break = MDLabel(text=' ', height='30sp', size_hint_y=None)
        self.list_view.add_widget(info_break)

    def lbl_info_app(self, l_text: str, r_text: str, r_color: str='Secondary'):
        layout_box = MDBoxLayout(
            orientation='horizontal',
            size_hint_x=None, size_hint_y=None,
            width='450sp', height='30sp'
        )
        lbl_left = MDLabel(
            text=l_text,
            halign='right',
            size_hint_x=None, size_hint_y=None,
            width='200sp', height='30sp',
            theme_text_color='Hint'
        )
        lbl_middle = MDLabel(
            text=' ',
            halign = 'center',
            size_hint_x=None, size_hint_y=None,
            width='10sp', height='30sp'
        )
        lbl_right = MDLabel(
            text=r_text,
            size_hint_y=None,
            height='30sp',
            theme_text_color=r_color
        )
        layout_box.add_widget(lbl_left)
        layout_box.add_widget(lbl_middle)
        layout_box.add_widget(lbl_right)
        self.list_view.add_widget(layout_box)

    def lbl_info_mdsanima(self, text_info: str, halign: str, height: str):
        info_mdsanima = MDLabel(
            text=text_info,
            halign=halign,
            width='512sp', height=height,
            padding_x=40, line_height=1.2,
            size_hint_x=None, size_hint_y=None,
            theme_text_color='Secondary'
        )
        self.list_view.add_widget(info_mdsanima)

